add hyphenate-limit-chars CSS property (#22747)
mdn/content: files/en-us/web/css/hyphenate-limit-chars/index.md

---
title: hyphenate-limit-chars
slug: Web/CSS/hyphenate-limit-chars
page-type: css-property
tags:
  - CSS
  - CSS Property
  - CSS Text
  - Reference
  - recipe:css-property
browser-compat: css.properties.hyphenate-limit-chars
---

{{CSSRef}}

> The **`hyphenate-limit-chars`** [CSS](/en-US/docs/Web/CSS) property provides fine-grained control over hyphenation in text, allowing for better typography by avoiding awkward hyphenations and setting appropriate hyphenation for different languages. It takes three numeric values — the minimum word length before words should be hyphenated, the minimum number of characters before the hyphen, and the minimum number of characters after the hyphen.

Reviewer: To keep this consistent with the description on other CSS property pages, how about we first say what the property does and then describe the advantages of having this property?

```suggestion
The **`hyphenate-limit-chars`** [CSS](/en-US/docs/Web/CSS) property specifies the minimum word length to allow hyphenation of words as well as the the minimum number of characters before and after the hyphen.

This property provides you with fine-grained control over hyphenation in text. This control enables you to avoid awkward hyphenations and set appropriate hyphenation for different languages, which, in turn, allows for better typography.
```

Author: Reads great, thanks Dipika!

{{EmbedInteractiveExample("pages/css/hyphenate-limit-chars.html")}}

> In the interactive example above, the default minimum word length setting of `hyphenate-limit-chars` is too large for either of the long words to be hyphenated, therefore it looks untidy. However, by choosing the other value option, that limit is brought down so that both of the long words are hyphenated, making for a neater layout.

Reviewer: ```suggestion
In the interactive example above, the default minimum word length setting of `hyphenate-limit-chars` is too large for either of the long words to be hyphenated, and so the output looks untidy. However, by choosing the other value option, the minimum word length limit is reduced. This allows both the long words to be hyphenated, making for a neater layout.
```

## Syntax

```css
/* Numeric values */
hyphenate-limit-chars: 10 4 4;
hyphenate-limit-chars: 10 4;
hyphenate-limit-chars: 10;

/* Keyword values */
hyphenate-limit-chars: auto auto auto;
hyphenate-limit-chars: auto auto;
hyphenate-limit-chars: auto;

/* Mix of numeric and keyword values */
hyphenate-limit-chars: 10 auto 4;
hyphenate-limit-chars: 10 auto;
hyphenate-limit-chars: auto 3;

/* Global values */
hyphenate-limit-chars: inherit;
hyphenate-limit-chars: initial;
hyphenate-limit-chars: revert;
hyphenate-limit-chars: revert-layer;
hyphenate-limit-chars: unset;
```

> The `hyphenate-limit-chars` property takes 1–3 values, which can be numeric or `auto`, as explained below.

Reviewer: ```suggestion
The `hyphenate-limit-chars` property takes 1–3 values that can be numeric or `auto`, as explained below.
```

### Values

- `x y z`
  - : The first value is the minimum word length before words should be hyphenated. The second value is the minimum number of characters before the hyphen. The third value is the minimum number of characters after the hyphen.
- `x y`
  - : The first value is the minimum word length before words should be hyphenated. The second value is the minimum number of characters before the hyphen. The minimum number of characters after the hyphen will be set equal to the second value.
- `x`
  - : The value is the minimum word length before words should be hyphenated. The minimum number of characters before and after the hyphen will be set to `auto`.

> If `auto` is set for any of the values, it means that the user agent will choose an appropriate value for the current layout. Unless the user agent can calculate a better value, the following default values are used:

Reviewer: ```suggestion
If `auto` is set for any of the values, the user agent will choose an appropriate value for the current layout. Unless the user agent can calculate a better value, the following default values will be used:
```

> - Minimum hyphenated word length: 5

Reviewer: ```suggestion
- Minimum word length to allow hyphenation: 5
```
